homeassistant/components/intent/timers.py

```diff
@@ -0,0 +1,772 @@
+        minutes: int | None,
+        seconds: int | None,
+        language: str,
+        device_id: str | None,
+        name: str | None = None,
+        data: dict[str, Any] | None = None,
+    ) -> str:
+        """Start a timer."""
+        total_seconds = 0
+        if hours is not None:
+            total_seconds += 60 * 60 * hours
+
+        if minutes is not None:
+            total_seconds += 60 * minutes
+
+        if seconds is not None:
+            total_seconds += seconds
+
+        timer_id = ulid.ulid_now()
+        created_at = time.monotonic_ns()
+        timer = TimerInfo(
+            id=timer_id,
+            name=name,
+            start_hours=hours,
+            start_minutes=minutes,
+            start_seconds=seconds,
+            seconds=total_seconds,
+            language=language,
+            device_id=device_id,
+            task=self.hass.async_create_background_task(
+                self._wait_for_timer(timer_id, total_seconds, created_at),
+                name=f"Timer {timer_id}",
+            ),
+            created_at=created_at,
+            updated_at=created_at,
+            data=data,
+        )
+
+        # Fill in area/floor info
+        if device_id and (device := self.device_registry.async_get(device_id)):
+            timer.area_id = device.area_id
+            if device.area_id and (
+                area := self.area_registry.async_get_area(device.area_id)
+            ):
+                timer.floor_id = area.floor_id
+
+        self.timers[timer_id] = timer
+
+        self.hass.bus.async_fire(EVENT_INTENT_TIMER_STARTED, timer.to_event())
+
+        _LOGGER.debug(
+            "Timer started: id=%s, name=%s, hours=%s, minutes=%s, seconds=%s, data=%s, device_id=%s",
+            timer_id,
+            name,
+            hours,
+            minutes,
+            seconds,
+            data,
+            device_id,
+        )
+
+        return timer_id
+
+    async def _wait_for_timer(
+        self, timer_id: str, seconds: int, updated_at: int
+    ) -> None:
+        """Sleep until timer is up. Timer is only finished if it hasn't been updated."""
+        try:
+            await asyncio.sleep(seconds)
+            if (timer := self.timers.get(timer_id)) and (
+                timer.updated_at == updated_at
+            ):
+                await self._timer_finished(timer_id)
+        except asyncio.CancelledError:
+            pass  # expected when timer is updated
+
+    async def cancel_timer(self, timer_id: str) -> None:
+        """Cancel a timer."""
+        timer = self.timers.pop(timer_id, None)
+        if timer is None:
+            return
+
+        # Capture event data before cancellation to get accurate seconds_left
+        event_data = timer.to_event()
+
+        timer.seconds = 0
+        timer.updated_at = time.monotonic_ns()
+        if not timer.is_paused:
+            timer.task.cancel()
+
+        self.hass.bus.async_fire(EVENT_INTENT_TIMER_CANCELLED, event_data)
```

Comment:
you always need to be careful about what we want to include in events. By including the full data, it now becomes an admin-only event. If it just changes the timer ID + action + attribte that was changed (if update), it can be accessible to all. This is how we do it for the registries.

Reply:
Moved to callbacks instead.